### ISSUE TITLE
Add support for Regex

### ISSUE THREAD

**Issue by Eshva** [johnnyreilly/jQuery.Validation.Unobtrusive.Native#11]
Your library doesn't support regex validation (RegularExpressionAttribute). Please add it.
Microsoft jQuery Unobtrusive Validation supports it.

**Comment by stefanb** [comment]
Please keep in mind that regex implementations might differ between environments (.NET vs JavaScript).

**Comment by stefanb** [comment]
This can be fixed by adding a "regex" validation method:
```javascript
	$.validator.addMethod(
        "regex",
        function (value, element, param) {
            var re = new RegExp(param.pattern);
            return this.optional(element) || re.test(value);
        },
        "Invalid format."
    );
```

**Comment by Zhaph** [comment]
I've submitted a pull request (#17) that adds support for the `pattern` flavour of Regex that's supplied with the official `additional-methods.js`, along with support for FileExtensions and the native MinLength and MaxLength attributes.

**Comment by johnnyreilly** [comment]
Resolved by #17
